src/requests/sign-request.ts

```diff
@@ -112,7 +110,7 @@ export const signRequest = (
 
   const path = getNormalizedEncodedURI(canonicalRequest.path)
   const method = canonicalRequest.method
-  const headers = canonicalRequest.headers ? normalizeHeaders(canonicalRequest.headers) : {}
+  const headers = normalizeHeaders(canonicalRequest.headers || {})
```

Comment:
Use `??` if you really want to do this, but I'd rather save the call if we don't get headers

Reply:
Ok I'll revert this, sorry for adding a style choice to a fix PR